packages/dds/tree/docs/main/compatibility.md

```diff
@@ -147,3 +147,82 @@ Snapshot tests are effective at catching changes which inadvertently modify the
+    OptionalFieldCodec-->RevisionTagCodec
+    SequenceFieldCodec-->RevisionTagCodec
+    NodeKeyFieldCodec-->RevisionTagCodec
+    ForbiddenFieldCodec-->RevisionTagCodec
+    OptionalFieldCodec-->FieldBatchCodec
+    SequenceFieldCodec-->FieldBatchCodec
+    NodeKeyFieldCodec-->FieldBatchCodec
+    ForbiddenFieldCodec-->FieldBatchCodec
+
+    classDef summarizable stroke-width:8px
+    classDef op stroke-width:8px
+    classDef versioned fill:#f96
+    class ForestCodec,EditManagerCodec,DetachedNodeToFieldCodec,SchemaCodec summarizable
+    class MessageCodec op
+    class FieldBatchCodec,SchemaCodec,ForestCodec,MessageCodec,EditManagerCodec,DetachedNodeToFieldCodec versioned
+```
+
+> Field kind codecs are actually corecursive with ModularChangeset's codec with respect to the encoded data. That doesn't significantly affect guidance around updating the persisted format.
+
+In this diagram, large borders represent 'top-level codecs', i.e. codecs which directly define the data format for a summary blob or op.
+Orange codecs provide explicit versions to the data they encode.
+
+Because all data is versioned at the top level, we can conceptually extend that version to include all other non-explicitly versioned containing data, even if that data isn't explicitly written by the same codec.
+For example, a format change in `SchemaChangeCodec` could be implemented by adding support for a new version on each of its nearest explicitly versioned consumers, i.e. `MessageCodec` and `EditManagerCodec`.
+This new version would use the same code for all bits of `MessageCodec`, `EditManagerCodec`, and `ChangeFamilyCodec`, but pass enough context down to `SchemaChangeCodec` to resolve to the newer format.
+In this mannger, the mapping between explicitly versioned data and implicitly versioned data for composed codecs is managed in code.
+
+## Current code guidelines
+
+Codecs which explicitly version their data should export a codec which takes in a write version and supports reading all supported versions.
+The write version will ultimately come from the user of SharedTree
+
+Codecs which do not explicitly version their data should export a codec family.
+This ensures that the consumer of the codec can use their versioning information to resolve the appropriate implicit version.
+
+Using the same example as above, under these guidelines `ChangeFamilyCodec` and `SchemaChangeCodec` should export codec families for composition purposes.
+To make a breaking change in `SchemaChangeCodec`,
+
+-   Add support for the new version in `SchemaChangeCodec`, adding it to the exposed codec family
+-   Add a new version for `ChangeFamilyCodec` which leverages the new `SchemaChangeCodec`
+-   Add a new version for `EditManagerCodec` which leverages the new `ChangeFamilyCodec`
+-   Add a new version for `MessageCodec` which leverages the new `ChangeFamilyCodec`
+-   Update `SharedTreeOptions` / `SharedTreeFormatVersion` with the new write version
```

Comment:
This may need clarification: is this "update" in the sense of "expose it so that users of ST can opt into it" or "update what ST will use be default"?